src/DotUtils.MsBuild.SensitiveDataDetector/ISensitiveDataDetector.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Build.SensitiveDataDetector;
+
+namespace DotUtils.MsBuild.SensitiveDataDetector
+{
+    public interface ISensitiveDataDetector
+    {
+        Dictionary<SensitiveDataKind, List<SecretDescriptor>> Detect(string input);
+    }
+
+    public record SecretDescriptor
```

Comment:
Very nitpicky nitpick :-D - this feels as an immutable and very compact data - so would be better to be `readonly record struct` (the setter would be removed at the same time)